src/semeio/fmudesign/design_distributions.py

```diff
@@ -315,58 +342,49 @@ def draw_values_pert(dist_parameters, numreals, rng, normalscoresamples=None):
+    high = float(dist_parameters[2])
+    scale = (
+        float(dist_parameters[3])
+        if len(dist_parameters) == 4
+        else 4  # pert 3 parameter distribution
+    )
+
+    if high == low:  # collapsed distribution
+        print(
+            "Low and high parameters for pert distribution"
+            " are equal. Using constant {}".format(low)
         )
         if normalscoresamples is not None:
-            if high == low:  # collapsed distribution
-                print(
-                    "Low and high parameters for pert distribution"
-                    " are equal. Using constant {}".format(low)
-                )
-                values = scipy.stats.uniform.ppf(
-                    scipy.stats.norm.cdf(normalscoresamples), loc=low, scale=0
-                )
-            else:
-                muval = (low + high + scale * mode) / (scale + 2)
-                if np.isclose(muval, mode):
-                    alpha1 = (scale / 2) + 1
-                else:
-                    alpha1 = ((muval - low) * (2 * mode - low - high)) / (
-                        (mode - muval) * (high - low)
-                    )
-                alpha2 = alpha1 * (high - muval) / (muval - low)
-                values = scipy.stats.beta.ppf(
-                    scipy.stats.norm.cdf(normalscoresamples), alpha1, alpha2
-                )
+            values = scipy.stats.uniform.ppf(
+                scipy.stats.norm.cdf(normalscoresamples), loc=low, scale=0
+            )
         else:
-            if high == low:  # collapsed distribution
-                print(
-                    "Low and high parameters for pert distribution"
-                    " are equal. Using constant {}".format(low)
-                )
-                distribution = scipy.stats.uniform(loc=low, scale=0)
-            else:
-                muval = (low + high + scale * mode) / (scale + 2)
-                if np.isclose(muval, mode):
-                    alpha1 = (scale / 2) + 1
-                else:
-                    alpha1 = ((muval - low) * (2 * mode - low - high)) / (
-                        (mode - muval) * (high - low)
-                    )
-                alpha2 = alpha1 * (high - muval) / (muval - low)
-                distribution = scipy.stats.beta(alpha1, alpha2)
-            values = distribution.rvs(size=numreals, random_state=rng)
+            values = np.full(numreals, low)
     else:
-        raise ValueError(msg)
-    # For pert distribution scale afterwards:
-    return values * (dist_parameters[2] - dist_parameters[0]) + dist_parameters[0]
+        muval = (low + high + scale * mode) / (scale + 2)
+        if np.isclose(muval, mode):
+            alpha1 = (scale / 2) + 1
+        else:
+            alpha1 = ((muval - low) * (2 * mode - low - high)) / (
+                (mode - muval) * (high - low)
+            )
+        alpha2 = alpha1 * (high - muval) / (muval - low)
+
+        if normalscoresamples is not None:
```

Comment:
perhaps this IF can be written as:

```python
uniform_samples = generate_stratified_samples(numreals, rng) if (normalscoresamples is None) else normalscoresamples
values = scipy.stats.beta.ppf(uniform_samples, alpha1, alpha2)
```

or pulling the `values` out of the IF:

```python
if normalscoresamples is not None:
    uniform_samples = normalscoresamples 
else:
    uniform_samples = generate_stratified_samples(numreals, rng)

values = scipy.stats.beta.ppf(uniform_samples, alpha1, alpha2)
```

Reply:
What about something like this:

```python
if normalscoresamples is not None:
    uniform_samples = scipy.stats.norm.cdf(normalscoresamples)
else:
    uniform_samples = generate_stratified_samples(numreals, rng)

values = scipy.stats.beta.ppf(uniform_samples, alpha1, alpha2)
```

Reply:
yes. i had a mistake in my comment where i forgot to use CDF. this looks correct and the code looks better with the common part outside of the IF

Reply:
Alright, I've added the last version.